test/frontend/test_node.py

```diff
@@ -38,13 +38,15 @@ class TestNode(unittest.TestCase):
   pass
 
 def create_test(test_data):
-  test_option = test_data[5] if len(test_data) > 5 else {}
+  test_option = test_data[7] if len(test_data) > 7 else {}
 
   def do_test_expected(self):
     tf_op = test_data[1]
     output_name = test_data[2]
     inputs = test_data[3]
     attrs = test_data[4]
+    device = test_data[5]
+    channel_last = test_data[6]
```

Comment:
Rather than specifying these two attributes, can we enumerate through all possibilities? For instance, if there's a GPU, then test NHWC and NCHW, if not, only test NHWC. Checks can be done via `supports_device` function.

This won't have any effects on CI, but I can test them locally on my work machine with GPU. This could significantly increase confidence in correctness.

Reply:
Ok, but with GPU, I think the only correct data format is NCHW, right ? 

Testing all possible data format more dynamically is a good idea.

Reply:
To quote from this thread https://github.com/tensorflow/tensorflow/issues/2660:

> MaxPool on GPU supports NCHW and NHWC
> MaxPool on CPU only supports NHWC.

I'd expect the same for most other ops that care about input formats.

Reply:
Can you try on a gpu to be really sur of that ?
Because it could impact my backend modifications in order to manage the data format...

Reply:
I'll let you know my trials later today but that statement was made by TF developers from Google so I'd consider that more authoritative than even my experimental results.